Automattic/newspack-custom-content-migrator
# feat(ContentDiff): migrate all taxonomies

## How to test
- Create a local site, and install a couple of plugins that use custom taxonomies ([Newspack Multibranded Site](https://github.com/automattic/newspack-multibranded-site), [Co-Authors Plus](https://wordpress.org/plugins/co-authors-plus/), ...).
- Add a couple posts, and link them to these custom taxonomies (setting post brands, and co-authors) as long as setting posts categories and tags.
1. Take a backup of the database (this will be our staging snapshot `staging.sql`).
2. Add more posts and custom taxonomies.
3. Rename all the database tables to use the prefix `live_` instead of `wp_`.
4. Import the backup from Step 1.
5. Create a logs folder somewhere in your computer `/tmp/test_cdiff/`.
6. Inside that folder create two files:
- An empty `content-diff__modified-ids.log` (unless if you updated some posts, then put their IDs in here in a CSV format)
- `content-diff__new-ids-csv.log` that contains the IDs of the new posts in a CSV format (p.s. if you're testing with CAP don't forget to add the co-authors entity IDs too because they are custom posts).
7. Run the migration command (Set the names of your custom taxonomies in the parameter `--custom-taxonomies-csv`):
`wp newspack-content-migrator content-diff-migrate-live-content --import-dir=/tmp/test_cdiff/ --live-table-prefix=live_ --custom-taxonomies-csv=author,brand`
8. Check that everything was migrated correctly.

---
